controller/controller.go

```diff
@@ -124,6 +125,9 @@ var unitExp = regexp.MustCompile("@.service")
 // 	 bar@2.service
 //
 func (r Request) ExtendSlices() (Request, error) {
+	if len(r.SliceIDs) == 0 {
+		return r, nil
+	}
```

Comment:
I guess this is the fix. I don't understand it though. The behavior of `ExtendSlices` was already what this change tries to achieve, IMO.